### DISCUSSION TITLE
aromatic label not removed during a reaction

### DISCUSSION THREAD

**Discussion by cottonkiet** [rdkit/rdkit#5268] [Q&A]
Hi

I am using the following reaction:
`[c:2]1[c:3][#8,#16;a:4][c:5][n:1]1>>O=[#6:2]-[#7:1](-[#6:5]=O)-[#6:3]=[#8,#16:4]`
with one reactant:
`CC(C)NC(=O)C1=COC(=N1)[C@@H](NC(=O)OC(C)(C)C)C(C)C`
but it produced a strange product
`CC(C)NC(=O)C(=O)N(C=o)C(=O)[C@@H](NC(=O)OC(C)(C)C)C(C)C`
In theory, it should produce a product
`CC(C)NC(=O)C(=O)N(C=O)C(=O)[C@@H](NC(=O)OC(C)(C)C)C(C)C`

It seems that the problem is related to atom 4 and it's somehow still labelled as aromatic in the product 

I tried to forced it to be aliphatic in the product side such as 
`[c:2]1[c:3][#8,#16;a:4][c:5][n:1]1>>O=[#6:2]-[#7:1](-[#6:5]=O)-[#6:3]=[#8,#16;A:4]`
but it does not work. 

somehow if I change to (removed the possibility of S for atom 4)
`[c:2]1[c:3][#8;a:4][c:5][n:1]1>>O=[#6:2]-[#7:1](-[#6:5]=O)-[#6:3]=[#8:4]`
it works. but this is strange that [#8,#16] won't work.  Does anyone know the reason or is this a potential bug? Thanks a lot!

To reproduce

```
from rdkit import Chem
from rdkit.Chem import AllChem
react = Chem.MolFromSmiles('CC(C)NC(=O)C1=COC(=N1)[C@@H](NC(=O)OC(C)(C)C)C(C)C')
display(react)
scheme = '[c:2]1[c:3][#8,#16;a:4][c:5][n:1]1>>O=[#6:2]-[#7:1](-[#6:5]=O)-[#6:3]=[#8,#16:4]'
rxn = AllChem.ReactionFromSmarts(scheme)
display(rxn)
prod = rxn.RunReactants([react])
len(prod)
display(prod)
```

Configuration (please complete the following information):

RDKit version: 2021.09.5
OS: Windows
Python version (if relevant): 3.9
Are you using conda? Yes
If you are using conda, which channel did you install the rdkit from? default

**Comment by wopozka** [comment]
@cottonkiet 
It looks direct transformations of type a->A, A->a are not supported in rdkit reactions, and RHS understands *, a, A as to "copy everyting". I have no idea whether it is a bug or feature. The solution is either to use two separate SMARTS for 's' and 'o' with proper definition on RHS, or postprocess reaction results and set proper flags to given atoms, according to your rules (eg. support a->A, A->a, transformations).

**Reply by cottonkiet** [reply]
Thank you! But what's special about [#8,#16] compared to [#8] about whether a->A is working or not?

**Reply by wopozka** [reply]
What I see (I did not look in the source code though) if there is a list on RHS, then it is equivalent of star, regardless what is there. You do not even need to have oxygen or sulfur there, you can have whatever and internally it is probably replaced with *.

**Reply by cottonkiet** [reply]
Thanks a lot! Then that looks like a bug. I will try to report it.

**Reply by wopozka** [reply]
I am not sure whether it is a bug. Constructions like lists do not have to much sense on RHS of the reaction, as LHS is responsible for mach and as you do not do nuclear reaction, you should receive the same atom on RHS.  Therefore if there is a list on LHS then on the other side, you should put an asterisk symbol, not a list. As rdkit reaction work on very low level it works very well for definitions without list, where you can precisely define the transformation. For problem you presented you need to define either two separate reaction SMARTS for oxygen and sulfur, or create a procedure which would apply necesary corrections to the molecule after the reaction. The last solution is used in commercial software.

**Reply by cottonkiet** [reply]
I see, so is it impossible for list atom to minic the way a certain atom will do on the level of rdkit reaction? If so, would you know on top of your head some post-reaction procedures for any reaction (not necessarily this one) for me to learn how to do these things? I assume I need to find the index the target atom which undergo the reaction? Especially when there are multiple reaction centers, it will be tough. Thanks a lot!

**Reply by wopozka** [reply]
Well, lets see if there is any reaction on bug report, but I recon it might be just the design. Answering your question. I am not able to give you a ready solution, because AFAIK there is no free available library or software doing this (I might be wrong though), but to give you some idea how to do it. As you noticed you need to know where was the match in the reaction. You can achieve this in two ways. You can either use isotopes for this purpose or to use rdkit features for that (it is briefly discussed there: https://github.com/rdkit/rdkit/issues/1269). In case of isotopes, you at the begining mark each atom of the molecule with different isotope number (starting from 1 to given number of atoms). Then you can easily find which atoms mach the reaction, find their isotope number, and as isotopes stay unchanged after the reaction, you can easily find corresponding atoms on the other side. Then it is up to you to decide what to do for given atoms - to support transformations you define (eg. if you see A on RHS then, you change given atom to aliphatic, if you see a then change to aromatic, etc). It would be advicable also to remove free radicals (they might be produce durinig reactions, due to the lack of hydrogens definition on RHS) and eventually sanitize molecule. This algo would work providing you do not have real isotopes in your substrates, if you have, then the second method is much better (in general it is much more versatile than isotopes). I hope what I wrote here makes some sence to you :D.

**Reply by cottonkiet** [reply]
Thanks so much for your detailed explanation ! I will try to make it works! Thanks again!
